invenio_app_ils/ill/api.py

```diff
@@ -102,7 +107,48 @@ def create(cls, data, id_=None, **kwargs):
         cls.build_resolver_fields(data)
         return super().create(data, id_=id_, **kwargs)
 
+    @preserve_fields(fields=["created_by_pid"])
     def update(self, *args, **kwargs):
         """Update record."""
         super().update(*args, **kwargs)
         self.build_resolver_fields(self)
+
+    def create_loan(
+        self, transaction_location_pid=None, transaction_user_pid=None
+    ):
+        """Create a new loan out of this borrowing request.
+
+        The newly created loan PID is added to the borrowing request and the
+        status is changed to "ON_LOAN".
+        It is the responsibility of the caller to ensure that the current
+        borrowing request does not already have a loan attached and the initial
+        status is valid.
+
+        :param transaction_location_pid: the location pid of the transaction
+            that will be stored in the loan. If not passed, the current one
+            will be used.
+        :param transaction_user_pid: the user pid of the transaction
+            that will be stored in the loan. If not passed, the current logged
+            in user id will be used.
+        :return: the PID and record loan
+        """
+        tloc_pid = (
+            transaction_location_pid
+            or current_app.config["ILS_DEFAULT_LOCATION_PID"]
+        )
+        transaction_user_pid = transaction_user_pid or current_user.id
+        if arrow.get(self["loan_end_date"]) < arrow.now():
+            raise ILLError("The loan end date cannot be in the past.")
+
+        item_pid = dict(type=self._pid_type, value=self["pid"])
```

Comment:
do you know if we only have this in ILS or at some point in invenio too?

Reply:
I don't think it will happen in Invenio for the time being...